ui/src/components/visualizations/MapLibre/index.tsx

```diff
@@ -80,16 +69,25 @@ const MapLibre: React.FC<Props> = ({ className, fileContent, fileType }) => {
 
   const handleFlyToSelectedFeature = useCallback(() => {
     if (mapRef.current && selectedFeature) {
-      const coords = selectedFeature.geometry?.coordinates;
-      if (
-        (coords && selectedFeature.geometry.type === "Point") ||
-        selectedFeature.geometry.type === "MultiPoint"
-      ) {
-        mapRef.current.flyTo({
-          center: coords,
-          zoom: 16,
-          duration: 300,
-        });
+      try {
+        if (
+          !selectedFeature.geometry ||
+          !selectedFeature.geometry.type ||
+          !selectedFeature.geometry.coordinates
+        ) {
+          console.error("Invalid selectedFeature geometry:", selectedFeature);
+          return;
+        }
```

Comment:
Isn't this kind of redundant?  If bbox throws an error due to selectedFeature not being adequate shouldn't the try catch catch the error? You also aren't doing this elsewhere when using bbox.